controllers/admin/AdminCartRulesController.php

```diff
@@ -171,9 +171,7 @@ public function postProcess()
                     $rule_group_id = 1;
                     if (is_array($rule_group_array)) {
                         // Empty for (with a ; at the end), that just find the first rule_group_id available in rule_group_array
-                        for ($rule_group_id = 1; in_array($rule_group_id, $rule_group_array); ++$rule_group_id) {
-                            42;
-                        }
+                        for (; in_array($rule_group_id, $rule_group_array); ++$rule_group_id);
```

Comment:
Should be done with a while for a better readability? :thinking:

Reply:
Looks good for transparency. But I have a better idea.
The logic of the code gives us The first incremental result that is not in the array.
Isn't this better? 
`$rule_group_id = end($rule_group_array) + 1;`

Reply:
But ... why making it simple when we can make it complicated ? 😁

Reply:
I think they thought about something like this:
```php
$rule_group_array = [1, 2, 3, 4, 5, 9, 10];
for (; in_array($rule_group_id, $rule_group_array); ++$rule_group_id);
// => $rule_group_id = 6;
```

Reply:
Otherwise an autoincrement field should have been used

Reply:
The very best code change would be something that, when we read it, we *immediately* understand *exactly* what is going on 😄. Here, with the current implementation, we are "I think it does this ..." ; "I think they wanted to do that ..." => such code is a bringer of complexity and bugs

Reply:
The whole legacy code is like that :sweat_smile: